memgpt/memory.py

```diff
@@ -16,96 +18,214 @@
+        """Representation of the memory in-context"""
+        section_strs = []
+        for section, module in self.memory.items():
+            section_strs.append(f'<{section} characters="{len(module)}/{module.limit}">\n{module.value}\n</{section}>')
+        return "\n".join(section_strs)
 
     def to_dict(self):
-        return {
-            "persona": self.persona,
-            "human": self.human,
+        """Convert to dictionary representation"""
+        return {key: value.dict() for key, value in self.memory.items()}
+
+
+class ChatMemory(BaseMemory):
+
+    def __init__(self, persona: str, human: str, limit: int = 2000):
+        self.memory = {
+            "persona": MemoryModule(name="persona", value=persona, limit=limit),
+            "human": MemoryModule(name="human", value=human, limit=limit),
         }
 
-    @classmethod
-    def load(cls, state):
-        return cls(state["persona"], state["human"])
-
-    def edit_persona(self, new_persona):
-        if self.persona_char_limit and len(new_persona) > self.persona_char_limit:
-            error_msg = f"Edit failed: Exceeds {self.persona_char_limit} character limit (requested {len(new_persona)})."
-            if self.archival_memory_exists:
-                error_msg = f"{error_msg} Consider summarizing existing core memories in 'persona' and/or moving lower priority content to archival memory to free up space in core memory, then trying again."
-            raise ValueError(error_msg)
-
-        self.persona = new_persona
-        return len(self.persona)
-
-    def edit_human(self, new_human):
-        if self.human_char_limit and len(new_human) > self.human_char_limit:
-            error_msg = f"Edit failed: Exceeds {self.human_char_limit} character limit (requested {len(new_human)})."
-            if self.archival_memory_exists:
-                error_msg = f"{error_msg} Consider summarizing existing core memories in 'human' and/or moving lower priority content to archival memory to free up space in core memory, then trying again."
-            raise ValueError(error_msg)
-
-        self.human = new_human
-        return len(self.human)
-
-    def edit(self, field, content):
-        if field == "persona":
-            return self.edit_persona(content)
-        elif field == "human":
-            return self.edit_human(content)
-        else:
-            raise KeyError(f'No memory section named {field} (must be either "persona" or "human")')
-
-    def edit_append(self, field, content, sep="\n"):
-        if field == "persona":
-            new_content = self.persona + sep + content
-            return self.edit_persona(new_content)
-        elif field == "human":
-            new_content = self.human + sep + content
-            return self.edit_human(new_content)
-        else:
-            raise KeyError(f'No memory section named {field} (must be either "persona" or "human")')
+    def core_memory_append(self, name: str, content: str) -> Optional[str]:

```

Comment:
How come `core_memory_apend` is in the child method and not the base method?

Reply:
The docstrings are specific to the `ChatMemory` class (they mention human/persona) so would need to be overriden by customized memory.